hoss/projection_utilities.py

```diff
@@ -203,13 +203,20 @@ def get_projected_x_y_extents(
     grid_lats, grid_lons = get_grid_lat_lons(  # pylint: disable=unpacking-non-sequence
         x_values, y_values, crs
     )
+
     geographic_resolution = get_geographic_resolution(grid_lons, grid_lats)
 
-    resolved_geojson = get_resolved_geojson(
+    resolved_points = get_resolved_geojson(
         geographic_resolution, shape_file=shape_file, bounding_box=bounding_box
     )
-
-    return get_x_y_extents_from_geographic_points(resolved_geojson, crs)
+    filtered_points = [
+        (
+            max(min(lon, np.nanmax(grid_lons)), np.nanmin(grid_lons)),
+            max(min(lat, np.nanmax(grid_lats)), np.nanmin(grid_lats)),
+        )
+        for lon, lat in resolved_points
+    ]
+    return get_x_y_extents_from_geographic_points(filtered_points, crs)
```

Comment:
I'm not sure you're getting what you want here.  If you are trying to clip the region to the grid extents,you will be successful when some of the resolved points are inside the bounds, but if all points are outside the grid_lats/lons I think you will just get two values repeated (I think, I need to test).

If you're clipping to a limit, look at numpy.clip for an efficient way to do that